readium/streamer/src/main/java/org/readium/r2/streamer/ParserAssetFactory.kt

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2023 Readium Foundation. All rights reserved.
+ * Use of this source code is governed by the BSD-style license
+ * available in the top-level LICENSE file of the project.
+ */
+
+package org.readium.r2.streamer
+
+import java.io.File
+import java.nio.charset.Charset
+import org.json.JSONObject
+import org.readium.r2.shared.asset.Asset
+import org.readium.r2.shared.error.*
+import org.readium.r2.shared.fetcher.ContainerFetcher
+import org.readium.r2.shared.fetcher.ResourceFetcher
+import org.readium.r2.shared.fetcher.RoutingFetcher
+import org.readium.r2.shared.publication.Link
+import org.readium.r2.shared.publication.Manifest
+import org.readium.r2.shared.publication.Publication
+import org.readium.r2.shared.resource.Container
+import org.readium.r2.shared.resource.Resource
+import org.readium.r2.shared.util.http.HttpClient
+import org.readium.r2.shared.util.http.HttpFetcher
+import org.readium.r2.shared.util.mediatype.MediaType
+import org.readium.r2.shared.util.mediatype.MediaTypeRetriever
+import org.readium.r2.streamer.parser.PublicationParser
+
+internal class ParserAssetFactory(
+    private val httpClient: HttpClient,
+    private val mediaTypeRetriever: MediaTypeRetriever,
+) {
+
+    suspend fun createParserAsset(
+        asset: Asset,
+    ): Try<PublicationParser.Asset, Publication.OpeningException> {
+        return when (asset) {
+            is Asset.Container ->
+                Try.success(createFetcherForContainer(asset.container, asset.mediaType, asset.name))
+            is Asset.Resource ->
+                createFetcherForResource(asset.resource, asset.mediaType, asset.name)
+        }
+    }
+
+    private fun createFetcherForContainer(
```

Comment:
The name of this method and the following looks incorrect?

Reply:
Legacy, thanks.